### ISSUE TITLE
New lemma-tizer plugin for ukrainian language.

### ISSUE THREAD

**Issue by mrgambal** [elastic/elasticsearch#18303]
Hi all,

I wonder whether you are interested in supporting a plugin which provides a mapping between ukrainian word forms and their lemmas. Some tests and docs go out-of-the-box =) .

https://github.com/mrgambal/elasticsearch-ukrainian-lemmatizer

It's really simple but still works and generates some value for its users.

P.S: @HonzaKral we had a discussion rgd this one on PyCon UA. Could you by any chance help?

**Comment by clintongormley** [comment]
Hi @mrgambal 

Thanks for the offer.  I wonder if it shouldn't go directly into Lucene instead - that's where all our other analysis components live.  Also, I don't see a license in the repository.  It would have to be a compatible license for us (or Lucene) to accept the code.

**Comment by mrgambal** [comment]
Hi @clintongormley 

Thanks for your response! Good spot on the license: I'll add a MIT, if you don't mind (hopefully - you don't).
IMO it isn't worth to be included into Lucene's set of plugins because this one doesn't provide any sort of real linguistic analysis. The thing is, it is just a `TokenFilter` and a huge dictionary of words and their lemmas.
Does that makes any sense?

**Comment by mrgambal** [comment]
Added a license in https://github.com/mrgambal/elasticsearch-ukrainian-lemmatizer/commit/e76a042218092e12be1642632e6ca7ff45b403fc

**Comment by clintongormley** [comment]
@rmuir what do you think?

**Comment by dchaplinsky** [comment]
@arysin, hooray!

**Comment by arysin** [comment]
@dchaplinsky Wow, great news!

**Comment by clintongormley** [comment]
I just asked @mikemccand and he would love to see this in Lucene.  Would you mind submitting this there instead?  Once it is merged into Lucene we can expose it in Elasticsearch

**Comment by mrgambal** [comment]
@clintongormley no objections from me. Where should I create a ticket?
I'm not sure whether I can put anythin' into their ASF tracker.

**Comment by clintongormley** [comment]
@mrgambal you should be able to create an issue here: https://issues.apache.org/jira/browse/LUCENE

**Comment by mrgambal** [comment]
@clintongormley here we go: https://issues.apache.org/jira/browse/LUCENE-7287
cc: @mikemccand 

Bear with me: it's my very first ASF ticket. If I did anything wrong - please let me know)

**Comment by clintongormley** [comment]
Closing in favour of https://issues.apache.org/jira/browse/LUCENE-7287

**Comment by dchaplinsky** [comment]
Hello @clintongormley . Thanks to @arysin now we have proper support for Ukrainian language in Lucene itself.

What should be done to propagate it to elasticsearch?

**Comment by clintongormley** [comment]
nice work @arysin 

@dchaplinsky we'll need to wait for Lucene 6.2 to be released.  I've opened this issue to track it: https://github.com/elastic/elasticsearch/issues/19433
